### PR TITLE
Dynamically import smol-toml package

### PR DESCRIPTION
**Pull request by heejaechang**
`module` change in `tsconfig.json` in https://github.com/microsoft/pyright/commit/6be5f8cfb989d0c77c313a1fd1a0286cc1b26c1e#diff-ac3a7b49798141e77c54e96bca892a082469cc83bb335322b35c01e020b060c8R1142 is causing issues in pylance due to it supporting multiple environments such as webpack, ts-node, browser and etc.

this is another approach to use `smol-toml`. confirmed it works as expected with webpack, ts-node and etc.

### CODE REVIEW

**Review by erictraut — CHANGES_REQUESTED** [review]
This change appears to break pyright. Refer to the mypy_primer output. All of the projects that use pyproject.toml are reporting errors now.

**Comment by heejaechang** [comment]
@erictraut can you take a look again?

File: packages/pyright-internal/src/common/shared.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+/*
+ * shared.ts
```
Comment (erictraut) [inline packages/pyright-internal/src/common/shared.ts:2]: Hmm, this isn't a very good name for a module. It's already located in a "common" directory, which implies that it's shared. The file name should describe what it does. Consider choosing a better name.

**Review by erictraut — APPROVED** [review]
Looks OK, but it would be better if the "shared.ts" file had a more appropriate and descriptive name.

File: packages/pyright-internal/src/common/shared.ts (hunk 1 above)
Comment (heejaechang) [reply packages/pyright-internal/src/common/shared.ts:2]: any suggestion? how about `sharedSetup.ts` ?

Comment (erictraut) [reply packages/pyright-internal/src/common/shared.ts:2]: How about `asyncInitialization.ts`?

**Pull request merged**